### PR TITLE
Update __array__ signatures with copy

### PR DESCRIPTION
**Pull request by Illviljan**
Align all `__array__`-methods according to the new signature. 
mypy can raise very hard to read errors if these are not correct, see #9237 for an example.

References: 
* https://numpy.org/doc/stable/user/basics.interoperability.html#dunder-array-interface
* https://github.com/numpy/numpy/blob/d031cac46a17dca63cd2dc3bb147d0d3fceaefe0/numpy/__init__.pyi#L1861-L1868

### CODE REVIEW

**Comment by keewis** [comment]
can you also remove the warning ignore in `pyproject.toml`, please?

File: xarray/core/indexing.py [hunk 1]
```diff
@@ -520,11 +526,6 @@ def get_duck_array(self):
         key = BasicIndexer((slice(None),) * self.ndim)
         return self[key]
 
-    def __array__(self, dtype: np.typing.DTypeLike = None) -> np.ndarray:
```
Comment (dcherian) [inline xarray/core/indexing.py:523]: This comment seems important... Why did you delete it?

Comment (dcherian) [reply xarray/core/indexing.py:523]: ah I see now.

🤞🏾

**Review by dcherian — APPROVED** [review]
Thanks for taking this on!
